app/src/main/java/com/gh4a/service/IssuesLabelService.java

```diff
@@ -0,0 +1,63 @@
+package com.gh4a.service;
+
+import android.app.Activity;
+import android.view.View;
+import com.gh4a.Gh4Application;
+import com.gh4a.R;
+import com.gh4a.utils.rx.RxTools;
+import org.eclipse.egit.github.core.Label;
+import org.eclipse.egit.github.core.RepositoryId;
+import org.eclipse.egit.github.core.service.LabelService;
+import java.net.URLEncoder;
+import io.reactivex.Observable;
+
+public class IssuesLabelService {
+    public static final int DELETE_ISSUE_LABEL = 0;
+    public static final int EDIT_ISSUE_LABEL = 1;
+    public static final int ADD_ISSUE_LABEL = 2;
```

Comment:
[Android Developers - Guide to App Architecture](https://developer.android.com/topic/libraries/architecture/guide.html#common_architectural_principles)
> The most important thing you should focus on is the separation of concerns in your app. It is a common mistake to write all your code in an Activity or a Fragment. Any code that does not handle a UI or operating system interaction should not be in these classes. Keeping them as lean as possible will allow you to avoid many lifecycle related problems. 

Just a question although, are Fragments shared among multiple activities in current architecture?

Reply:
> Just a question although, are Fragments shared among multiple activities in current architecture?

No, but fragments often have a shorter lifetime than their host activities.

Also your link doesn't seem applicable here. The loaders *are* a system API, thus operating system interaction.

Reply:
I was trying to make the point that including **150 lines** of code, all concerning **network calls**, would make Activity and Fragment code harder to read ... 

Furthermore if **fragments** are not shared among multiple **activities**, wouldn't it be easier to lean more on **packages** and use a **per-screen** structure, like creating **one package per screen**, that would contain both: 
- An Activity
- Its fragment(s)
- Its Service file 

**Screens packages** could than be regrouped by **theme**

Reply:
Perhaps, but that's not a topic for this pull request. @maniac103 has a point here. If an activity used multiple services, each with its own loader ids, then we would easily encounter loader id collision and unexpected behavior. For example `RepositoryService.LOAD_README` and `IssuesLabelService.DELETE_ISSUE_LABEL` both equal 0. I fear that this would break if both were used from same activity/fragment.

Reply:
> For example RepositoryService.LOAD_README and IssuesLabelService.DELETE_ISSUE_LABEL both equal 0. I fear that this would break if both were used from same activity/fragment.

Yes, that's exactly what I meant.

> I was trying to make the point that including 150 lines of code, all concerning network calls, would make Activity and Fragment code harder to read ...

I disagree with that. IMHO overusing modularization can actually impair code readability. Putting half of the code of a method like loadReadme() into a separate class just doesn't seem useful to me.

Anyway, I think the ultimate goal should be converting the API bindings to use Rx anyway, at which point most of this discussion will be moot.